hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+
+One of the most common differences between target devices is the requirement for a specific SPI clock phase or polarity, CPOL and CPHA, which were described in the previous section [SPI Protocol Basics](#spi-protocol-basics).
+These clock parameters can be set via the {{< regref "CONFIG.CPOL">}} or {{< regref "CONFIG.CPHA" >}} register fields.
+Likewise, as also described in the previous section, if device setup times require a full clock cycle before sampling the output, Full-Cycle Mode can be enabled by asserting the {{< regref "CONFIG.FULLCYC" >}} bit.
+
+#### Clock rate selection
+
+The SPI clock rate for each peripheral is set by two factors:
+- The SPI_HOST core clock (which may be generated independently from the TL-UL iterface clock)
+- An additional 16-bit clock divider
+
+The SPI protocol usually requires activity (either sampling or asserting data), and either edge of the `sck` clock.
+For this reason the maximum `sck` frequency is at most one half the SPI_HOST core frequency.
+In order to support a broader range SPI clock frequencies, the SPI_HOST core operates on a separate clock domain, which may be independent from the TL-UL bus clock.
+This arrangement allows gives more flexibility in setting the serial clock frequency by means of an external PLL, and in principle the sck frequency could even be configured to *exceed* the bus clock.
+For example, if the TL-UL bus is operating at 100MHz, a 200MHz SPI core clock can be fed to the SPI_HOST IP, in order to acheive data rates of 100 MTransfers/s (near the maximum clock rate of the Winbond flash device).
+
+Since some peripheral devices attached to the same SPI_HOST may require different clock frequencies, there is also the option to divide the core clock by an additional factor when dealing with slower peripherals.
+
+$$T_\textrm{sck}=\frac{1}{2}\frac{T_\textrm{core}}{\textrm{CONFIG.CLKDIV}+1}$$
+
+Alternatively, this clock-divide feature can also be used to entirely bypass the need for an independent core clock.
+Instead the core can be optionally driven by the TL-UL bus clock, and the `sck` period can be adjust using the {{< regref CONFIG.CLKDIV >}} setting.
+
+#### Chip-select timing control
+
+Typically the `cs_n` line is automatically deasserted after the last edge of `sck`.
+However,  by asserting {{< regref "CONFIG.CSAAT" >}}, before issuing a particular command, one can instruct the core to hold `cs_n` low indefinitely after the last clock edge.
+This is for instance useful in merging two adjacent commands together, for creating very long commands such as continuous read operations.
+The `cs_n` line can then be deasserted by either issuing another command after clearing the {{< regref "CONFIG.CSAAT" >}} field, issuing a command to a different CSID, or simply resetting the core FSM via the {{< regref "CONTROL.RST_FSM">}}.
+
+Most devices require at least one-half `sck` clock-cycle between the edges of `cs_n` and between either edge of `cs_n` and the nearest `sck` edge.
```

Comment:
It's not really that mangled, because you do need to have 1/2 cycle between the rising edge of cs_n and the falling edge of cs_n.  Though I'm happy to remove that clause and leave that implicit. The main point of the paragraph stands.

Reply:
clause removed.